rmw_connextdds_common/src/common/rmw_impl.cpp

```diff
@@ -922,6 +924,31 @@ RMW_Connext_Publisher::assert_liveliness()
   return RMW_RET_OK;
 }
 
+rmw_ret_t
+RMW_Connext_Publisher::wait_for_all_acked(rmw_time_t wait_timeout)
+{
+  DDS_Duration_t timeout;
+
+  // TODO(Barry): While rmw_connextdds_duration_from_ros_time() is changed to the public function,
+  // replace below codes.
+  if (rmw_time_equal(wait_timeout, RMW_DURATION_INFINITE)) {
+    timeout = DDS_DURATION_INFINITE;
+  } else {
+    rmw_time_t clamped_time = rmw_dds_common::clamp_rmw_time_to_dds_time(wait_timeout);
+    timeout.sec = static_cast<DDS_Long>(clamped_time.sec);
+    timeout.nanosec = static_cast<DDS_UnsignedLong>(clamped_time.nsec);
+  }
+
+  switch (DDS_DataWriter_wait_for_acknowledgments(this->dds_writer, &timeout)) {
```

Comment:
Also, as a "coding style", I would prefer to cache the return code to a local variable, so we can also log it for the user (see below).
```suggestion
  const DDS_ReturnCode_t dds_rc =
    DDS_DataWriter_wait_for_acknowledgments(this->dds_writer, &timeout);
  switch (dds_rc) {
```